datafusion/src/physical_plan/file_format/mod.rs

```diff
@@ -293,6 +293,36 @@ impl PartitionColumnProjector {
         }
     }
 
+    // Creates a RecordBatch with values from the partition_values. Used when no non-partition values are read
+    fn project_empty(
```

Comment:
what about `project_from_size` instead?

Reply:
That's probably a better name project_empty doesn't really describe the method.